chain/index/events.go

```diff
@@ -236,48 +238,98 @@ func loadExecutedMessages(ctx context.Context, cs ChainStore, recomputeTipSetSta
 	return ems, nil
 }
 
-// checkTipsetIndexedStatus verifies if a specific tipset is indexed based on the EventFilter.
-// It returns nil if the tipset is indexed, ErrNotFound if it's not indexed or not specified,
-func (si *SqliteIndexer) checkTipsetIndexedStatus(ctx context.Context, f *EventFilter) error {
-	var tipsetKeyCid []byte
-	var err error
-
-	// Determine the tipset to check based on the filter
-	switch {
-	case f.TipsetCid != cid.Undef:
-		tipsetKeyCid = f.TipsetCid.Bytes()
-	case f.MinHeight >= 0 && f.MinHeight == f.MaxHeight:
-		tipsetKeyCid, err = si.getTipsetKeyCidByHeight(ctx, f.MinHeight)
-		if err != nil {
-			if err == ErrNotFound {
-				// this means that this is a null round and there exist no events for this epoch
-				return nil
-			}
```

Comment:
you've lost all of this logic in the process of refactoring, so we have a couple of problems now:

1. We're not testing the case where the filter specifies a specific `TipsetCid` - which we can do directly without needing to look at the range (case 1 above)
2. We're not taking the shortcut path for the case where the filter specifies a min==max, that one is easy because it's just one tipset to test (case 2 above)
3. Your new code just jumps straight in and assumes that `f.MinHeight` and `f.MaxHeight` are useful, but that may not be the case. They're allowed to have `-1` values, which signifies that neither have been specified (we do that in https://github.com/filecoin-project/lotus/blob/12d76bd9f11b14f0c89c69784616a5e9cb120058/node/impl/full/eth.go#L1844). Which is why the `>= 0` check existed in the original case.

So I think you'll need to restore some form of the original `switch` to accommodate those two cases above and then for the third case where `f.MinHeight >= 0 && f.MaxHeight >= 0`, and maybe `&& f.MinHeight < f.MaxHeight` just for good measure, just to be defensive about possible inputs.